jaeger-apachehttpclient/src/test/java/com/uber/jaeger/httpclient/JaegerRequestAndResponseInterceptorIntegrationTest.java

```diff
@@ -0,0 +1,124 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.uber.jaeger.httpclient;
+
+import static org.junit.Assert.assertEquals;
+
+import com.uber.jaeger.Span;
+import com.uber.jaeger.Tracer;
+import com.uber.jaeger.context.TraceContext;
+import com.uber.jaeger.context.TracingUtils;
+import com.uber.jaeger.reporters.InMemoryReporter;
+import com.uber.jaeger.samplers.ConstSampler;
+import com.uber.jaeger.samplers.Sampler;
+import org.apache.http.HttpHost;
+import org.apache.http.concurrent.FutureCallback;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.message.BasicHttpRequest;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockserver.client.server.MockServerClient;
+import org.mockserver.junit.MockServerRule;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JaegerRequestAndResponseInterceptorIntegrationTest {
+
+  @Rule public MockServerRule mockServerRule = new MockServerRule(this);
+  private InMemoryReporter reporter;
+  private Tracer tracer;
+
+  @SuppressWarnings("unused")
+  private MockServerClient mockServerClient;
+
+  private String BAGGAGE_KEY = "Bag_End";
+  private String BAGGAGE_VALUE = "smial_at_the_end_of_Bagshot_Row";
+
+  @Before
+  public void setUp() {
+    mockServerClient
+        .when(HttpRequest.request().withMethod("GET").withPath("/testing"))
+        .respond(HttpResponse.response().withStatusCode(200));
+    reporter = new InMemoryReporter();
+    Sampler sampler = new ConstSampler(true);
+    tracer = new Tracer.Builder("test_service", reporter, sampler).build();
+  }
+
+  @Test
+  public void testRequestAndResponseInterceptor() throws Exception {
+    //Set up a parent span context
+    TraceContext parentTraceContext = TracingUtils.getTraceContext();
+    Span parentSpan = (Span) tracer.buildSpan("parent_operation").start();
+    parentSpan.setBaggageItem(BAGGAGE_KEY, BAGGAGE_VALUE);
+    parentSpan.finish();
+    parentTraceContext.push(parentSpan);
+
+    CloseableHttpAsyncClient client =
+        HttpAsyncClients.custom()
+            .addInterceptorFirst(new JaegerRequestInterceptor(tracer))
+            .addInterceptorFirst(new JaegerResponseInterceptor())
```

Comment:
I would write a util function (e.g. in class `TracingInterceptors`) instead that takes a builder and attaches both interceptors. You can also make the interceptors themselves private to the package and only expose them via static factory methods in that class.